pandas/tests/indexing/test_multiindex.py

```diff
@@ -428,6 +428,50 @@ def test_xs_multiindex(self):
         expected.columns = expected.columns.droplevel('lvl1')
         tm.assert_frame_equal(result, expected)
 
+    def test_set_level_checkall(self):
+
+        idx = MultiIndex.from_tuples([(1, u'one'), (1, u'two'),
+                                     (2, u'one'), (2, u'two')],
+                                     names=['foo', 'bar'])
+        result = idx.set_levels([['a', 'b'], [1, 2]])
+        expected = MultiIndex(levels=[[u'a', u'b'], [1, 2]],
+                              labels=[[0, 0, 1, 1], [0, 1, 0, 1]],
+                              names=[u'foo', u'bar'])
+        tm.assert_index_equal(result, expected)
+
+        result = idx.set_levels(['a', 'b'], level=0)
```

Comment:
Now that we have these tests passing, it may be good to split these up into multiple tests so it's easier to review.